packages/core/src/components/Button/Button.tsx

```diff
@@ -46,6 +46,8 @@ export interface HvButtonProps
   classes?: HvButtonClasses;
   /** Whether the Button is selected or not. */
   selected?: boolean;
+  /** The component to be used as the root node. Defaults to `button`. */
+  component?: React.ElementType<React.HTMLAttributes<HTMLElement>>;
```

Comment:
Will this work for custom components (ie `Link`)? 🤔
Also, will we be able to  pass an `href` in `...others`? This `extends React.ButtonHTMLAttributes`.

For now, we could also extend from `Record<string, any>`, but ideally `HvButton` should inherit the `component`'s props.

Reply:
No, it will not. I'll review this, thanks!